components/ChallengeMaterial.tsx

```diff
@@ -96,20 +98,39 @@ export const ChallengeTitleCard: React.FC<ChallengeTitleCardProps> = ({
 export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
   currentChallenge
 }) => {
+  const { data } = React.useContext(SessionContext)
+
   return (
-    <div className="card shadow-sm border-0">
-      <div className="card-body challenge-question">
-        <h1 data-testid="challenge-question-title" className="card-title">
-          {currentChallenge.title}
-        </h1>
-        <div
-          data-testid="challenge-question-description"
-          className="bg-light p-3 mt-3"
-        >
-          <Markdown>{currentChallenge.description}</Markdown>
+    <>
+      <div className="card shadow-sm border-0">
+        <div className="card-body challenge-question">
+          <h1 data-testid="challenge-question-title" className="card-title">
+            {currentChallenge.title}
+          </h1>
+          <div
+            data-testid="challenge-question-description"
+            className="bg-light p-3 mt-3"
+          >
+            <Markdown>{currentChallenge.description}</Markdown>
+          </div>
         </div>
       </div>
-    </div>
+
+      {currentChallenge.submission?.diff && (
```

Comment:
I know `?` is cool, but it is error prone, especially if used in multiple places, so it is a practice we discourage. In long nested objects, you must put `?` behind every `.`, which potentially leads to more errors.

Please create variables instead.
```
const diff = _.get(currentChallenge, 'submission.diff')
const username = _.get(data, 'userInfo.username')
```

Reply:
will do! thanks :)

Reply:
cool I didn't know about `_.get()`